pkg/index/index.go

```
@@ -31,7 +30,14 @@ func NewTreeIndex() Index {
 	}
 }
 
-func (ti *treeIndex) Put(key []byte, rev Revision) {
+// Put inserts a new index entry
+// for now, assuming no error yield within this method
```

There is a panic instead of errors to follow fail early patterns.